README.md
# https-www.pythonanywhere.com-user-francescoalfano-consoles-9971451-


history.txt
    1  ls
    2  git config --global user.name "francescoalfano"
    3  git config --global user.email "samrotuna71@gmail.com"
    4  ls
    5  git config --global user.name "francescoalfano"
    6  cd mysite
    7  git status
    8  git add .
    9  git commit -m "les derniers sont les premiers. vous etes enseignant en droit publique. les connaissances en tant que telles."
   10  mkvirtualenv flask-tutorial --python=python3.6
   11  pip install flask flask-login flask-sqlalchemy mysql-connector-python
   12  pip freeze > requirements.txt
   13  git status
   14  git add .
   15  git commit -m "le onzieme ou le douzieme, le combien de fils de jacob"
   16  pip install Flask-Migrate
   17  export FLASK_APP=flask_app.py
   18  flask db init
   19  flask db migrate
   20  flask db stamp head
   21  pip freeze > requirements.txt
   22  git status
   23  git add .
   24  git commit -m "la cuisine des apotres"
   25  flask db migrate
   26  flask db upgrade
   27  git status
   28  git add .
   29  git commit -m "filles de la charite, c'etait notre dame de la chaume"
   30  flask db migrate
   31  flask db upgrade
   32  flask shell
   33  echo "# https-www.pythonanywhere.com-user-francescoalfano-consoles-9971451-" >> README.md
   34  git init
   35  git add README.md
   36  git commit -m "first commit"
   37  git remote add origin https://github.com/samrotuna711/https-www.pythonanywhere.com-user-francescoalfano-consoles-9971451-.git
   38  git push -u origin master
   39  history
   40  history > history.txt


flask_app.py
from datetime import datetime
from flask import Flask, redirect, render_template, request, url_for
from flask_login import current_user, login_required, login_user, LoginManager, logout_user, UserMixin
from flask_migrate import Migrate
from flask_sqlalchemy import SQLAlchemy
from werkzeug.security import check_password_hash


app = Flask(__name__)
app.config["DEBUG"] = True

SQLALCHEMY_DATABASE_URI = "mysql+mysqlconnector://{username}:{password}@{hostname}/{databasename}".format(
    username="francescoalfano",
    password="Exeter22",
    hostname="francescoalfano.mysql.pythonanywhere-services.com",
    databasename="francescoalfano$comments",
)
app.config["SQLALCHEMY_DATABASE_URI"] = SQLALCHEMY_DATABASE_URI
app.config["SQLALCHEMY_POOL_RECYCLE"] = 299
app.config["SQLALCHEMY_TRACK_MODIFICATIONS"] = False
db = SQLAlchemy(app)
migrate = Migrate(app, db)

app.secret_key = "5646545623112312987987987"
login_manager = LoginManager()
login_manager.init_app(app)


class User(UserMixin, db.Model):

    __tablename__ = "users"

    id = db.Column(db.Integer, primary_key=True)
    username = db.Column(db.String(128))
    password_hash = db.Column(db.String(128))

    def check_password(self, password):
        return check_password_hash(self.password_hash, password)


    def get_id(self):
        return self.username


@login_manager.user_loader
def load_user(user_id):
    return User.query.filter_by(username=user_id).first()


class Comment(db.Model):

    __tablename__ = "comments"

    id = db.Column(db.Integer, primary_key=True)
    content = db.Column(db.String(4096))
    posted = db.Column(db.DateTime, default=datetime.now)
    commenter_id = db.Column(db.Integer, db.ForeignKey('users.id'), nullable=True)
    commenter = db.relationship('User', foreign_keys=commenter_id)


@app.route("/", methods=["GET", "POST"])
def index():
    if request.method == "GET":
        return render_template("main_page.html", comments=Comment.query.all())

    if not current_user.is_authenticated:
        return redirect(url_for('index'))

    comment = Comment(content=request.form["contents"], commenter=current_user)
    db.session.add(comment)
    db.session.commit()
    return redirect(url_for('index'))



@app.route("/login/", methods=["GET", "POST"])
def login():
    if request.method == "GET":
        return render_template("login_page.html", error=False)

    user = load_user(request.form["username"])
    if user is None:
        return render_template("login_page.html", error=True)

    if not user.check_password(request.form["password"]):
        return render_template("login_page.html", error=True)

    login_user(user)
    return redirect(url_for('index'))



@app.route("/logout/")
@login_required
def logout():
    logout_user()
    return redirect(url_for('index'))

migrations/versions/e85258803044_.py
"""empty message

Revision ID: e85258803044
Revises: 49d9db3d2c50
Create Date: 2018-08-26 23:51:31.799647

"""
from alembic import op
import sqlalchemy as sa


# revision identifiers, used by Alembic.
revision = 'e85258803044'
down_revision = '49d9db3d2c50'
branch_labels = None
depends_on = None


def upgrade():
    # ### commands auto generated by Alembic - please adjust! ###
    op.create_table('users',
    sa.Column('id', sa.Integer(), nullable=False),
    sa.Column('username', sa.String(length=128), nullable=True),
    sa.Column('password_hash', sa.String(length=128), nullable=True),
    sa.PrimaryKeyConstraint('id')
    )
    op.add_column('comments', sa.Column('commenter_id', sa.Integer(), nullable=True))
    op.add_column('comments', sa.Column('posted', sa.DateTime(), nullable=True))
    op.create_foreign_key(None, 'comments', 'users', ['commenter_id'], ['id'])
    # ### end Alembic commands ###


def downgrade():
    # ### commands auto generated by Alembic - please adjust! ###
    op.drop_constraint(None, 'comments', type_='foreignkey')
    op.drop_column('comments', 'posted')
    op.drop_column('comments', 'commenter_id')
    op.drop_table('users')
    # ### end Alembic commands ###


migrations/versions/49d9db3d2c50_.py
"""empty message

Revision ID: 49d9db3d2c50
Revises: 
Create Date: 2018-08-26 23:27:14.545181

"""
from alembic import op
import sqlalchemy as sa


# revision identifiers, used by Alembic.
revision = '49d9db3d2c50'
down_revision = None
branch_labels = None
depends_on = None


def upgrade():
    # ### commands auto generated by Alembic - please adjust! ###
    op.create_table('users',
    sa.Column('id', sa.Integer(), nullable=False),
    sa.Column('username', sa.String(length=128), nullable=True),
    sa.Column('password_hash', sa.String(length=128), nullable=True),
    sa.PrimaryKeyConstraint('id')
    )
    op.create_table('comments',
    sa.Column('id', sa.Integer(), nullable=False),
    sa.Column('content', sa.String(length=4096), nullable=True),
    sa.Column('posted', sa.DateTime(), nullable=True),
    sa.Column('commenter_id', sa.Integer(), nullable=True),
    sa.ForeignKeyConstraint(['commenter_id'], ['users.id'], ),
    sa.PrimaryKeyConstraint('id')
    )
    # ### end Alembic commands ###


def downgrade():
    # ### commands auto generated by Alembic - please adjust! ###
    op.drop_table('comments')
    op.drop_table('users')
    # ### end Alembic commands ###
